client/src/templates/Challenges/redux/completion-epic.js

```diff
@@ -100,10 +99,7 @@ function postChallenge(update, username) {
 function submitModern(type, state) {
   const challengeType = state.challenge.challengeMeta.challengeType;
   const tests = challengeTestsSelector(state);
-  if (
-    hasNoTests(challengeType) ||
-    (tests.length > 0 && tests.every(test => test.pass && !test.err))
-  ) {
+  if (tests.every(test => test.pass && !test.err)) {
```

Comment:
This will return `true` when `tests` is an empty array - is that desired?

Reply:
It's the same, but it's more declarative.

`if (tests.length > 0 && tests.every...)` reads like I would think "if there are tests and they all pass, then..."

`if(test.every...)` makes "if there are tests" implicit. It's not a big deal, but that's the reasoning.

> This is not a feature.

Sorry, yeah, that's true. There probably should be one, though. I don't think it's good that we can forget to add tests and have that challenge default to being submittable.

Anyways, that's a problem for another day.